docs/data/joy/components/menu/MenuListComposition.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import PopperUnstyled from '@mui/base/PopperUnstyled';
+import ClickAwayListener from '@mui/base/ClickAwayListener';
+import { styled } from '@mui/joy/styles';
+import Button from '@mui/joy/Button';
+import MenuList from '@mui/joy/MenuList';
+import MenuItem from '@mui/joy/MenuItem';
+
+const Popup = styled(PopperUnstyled)({
+  zIndex: 1000,
+});
+
+export default function MenuListComposition(): JSX.Element {
```

Comment:
Since it's for the docs, I think we should keep the type simpler, even if slower:

```suggestion
export default function MenuListComposition() {
```

It's what the reset of the docs is doing, fixed in #38903.